### PR TITLE
Add a function to convert `dat` files to `CSV` format

### PR DESCRIPTION
**Pull request by Saransh-cpp**
I frequently use the data that `DeepXDE` outputs by converting it into `CSV` format, as it makes it compatible with libraries like `csv` and `pandas`. I think others might also find this useful (if they are already doing this with their own code) -
- Added a function to convert `dat` files into `CSV` files.
- Ran `black` across the codebase.
- Added some minor documentation

Usage -
```py
dde.dat_to_csv("../dat_data/heat_1D_loss.dat", "./heat_1D_loss_csv_file.csv", ["x", "t", "u_true", "u_pred"])
```

### CODE REVIEW

**Comment by lululxvi** [comment]
Great. Why there are so many formating changes in other files?

**Comment by Saransh-cpp** [comment]
I ran `black` for the entire `DeepXDE` folder. Should I revert them back?

**Comment by lululxvi** [comment]
Yes, that is better, to make this PR more clear.

File: deepxde/postprocessing.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import os
+import csv
```
Comment (lululxvi) [inline deepxde/postprocessing.py:2]:
```python
import csv
import os
```

File: deepxde/__init__.py [hunk 2]
```diff
@@ -22,7 +22,10 @@
     IC,
 )
 from .model import Model
-from .postprocessing import saveplot
+from .postprocessing import (
```
Comment (lululxvi) [inline deepxde/__init__.py:25]: It seems to me that `dat_to_csv` is not important enough to be directly under dde namespace

**Comment by lululxvi** [comment]
Last idea: should `dat_to_csv` in `postprocessing` or [`utils`](https://github.com/lululxvi/deepxde/blob/master/deepxde/utils/external.py)?

**Comment by Saransh-cpp** [comment]
> Last idea: should `dat_to_csv` in `postprocessing` or [`utils`](https://github.com/lululxvi/deepxde/blob/master/deepxde/utils/external.py)?

Yes, moving it `utils` sounds better!

**Pull request merged**